### PR TITLE
Make third party deps tool available to use outside startup-os

### PR DESCRIPTION
**Pull request by val-fed**

### CODE REVIEW

File: tools/build_file_generator/third_party_deps/ThirdPartyDepsAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
+/*
```
Comment (oferb) [inline tools/build_file_generator/third_party_deps/ThirdPartyDepsAnalyzer.java]: Did you change this code, or just move it?

File: tools/build_file_generator/third_party_deps/third_party_deps_updater.sh [hunk 2]
```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env bash
```
Comment (oferb) [inline tools/build_file_generator/third_party_deps/third_party_deps_updater.sh]: How did we do this before this script?

File: tools/build_file_generator/third_party_deps/ThirdPartyDepsAnalyzer.java (hunk 1 above)
Comment (val-fed) [reply tools/build_file_generator/third_party_deps/ThirdPartyDepsAnalyzer.java]: The code has lots of changes. I created draft PR to see all them - https://github.com/google/startup-os/pull/557/files

File: tools/build_file_generator/third_party_deps/third_party_deps_updater.sh (hunk 2 above)
Comment (val-fed) [reply tools/build_file_generator/third_party_deps/third_party_deps_updater.sh]: In the beginning I used `bazel query 'attr("visibility", "//visibility:public", //third_party/...)'` command and fill in deps in the BUILD file manually. 
We didn't have automation way for updating the BUILD file after adding/removing a third-party dependency in the project.

File: tools/build_file_generator/third_party_deps/third_party_deps_updater.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+# A script for updating third party dependencies inside third_party_deps_tool java_binary
+
+readonly REPO_ABS_PATH=$(git rev-parse --show-toplevel)
+# TODO: Make the getting of the path independent of the project
+readonly BUILD_FILE_ABS_PATH="$REPO_ABS_PATH/tools/build_file_generator/third_party_deps/BUILD"
+
+# Getting public third-party target names
+third_party_folder_path="$REPO_ABS_PATH/third_party"
```
Comment (oferb) [inline tools/build_file_generator/third_party_deps/third_party_deps_updater.sh]: Why is this not `readonly` like the others, and styles as a constant?

File: tools/build_file_generator/third_party_deps/third_party_deps_updater.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+declare -a actual_third_party_targets
```
Comment (vmax) [inline tools/build_file_generator/third_party_deps/third_party_deps_updater.sh]: @val-fed 
```suggestion
actual_third_party_targets=()
```

File: tools/build_file_generator/third_party_deps/third_party_deps_updater.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+for entry in $(bazel query 'attr("visibility", "//visibility:public", //third_party/...)'); do
+  actual_third_party_targets=("${actual_third_party_targets[@]}" "$entry")
```
Comment (vmax) [inline tools/build_file_generator/third_party_deps/third_party_deps_updater.sh]: @val-fed 
```suggestion
  actual_third_party_targets+=("$entry")
```

File: tools/build_file_generator/third_party_deps/third_party_deps_updater.sh [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+done
+
+# Deleting old targets
+sed -i '/# %START%/,/# %END%/{//!d}' ${BUILD_FILE_ABS_PATH}
```
Comment (vmax) [inline tools/build_file_generator/third_party_deps/third_party_deps_updater.sh]: @val-fed
There are couple of concerns here:
- `sed`, to be honest, is pretty cryptic to read. Maybe explain in detail what `sed` is meant to do here in terms of text manipulation ("delete text between %START and %END, while preserving the markers"). This way we'll be able to think of a better way to achieve the same goal.
- `-i <command> <file>` will only work on GNU sed. To make it portable you should do `sed -i.bak -e command file && rm file.bak`. See [here](https://stackoverflow.com/questions/5694228/sed-in-place-flag-that-works-both-on-mac-bsd-and-linux).
- Also, your version doesn't work on BSD sed 😉. Please add a semicolon at the end so it does not fail with `extra characters at the end of d command`: 
```suggestion
sed -i '/# %START%/,/# %END%/{//!d;}' ${BUILD_FILE_ABS_PATH}
```

File: tools/build_file_generator/third_party_deps/third_party_deps_updater.sh [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+# Adding new targets
+for target_to_add in "${actual_third_party_targets[@]}"; do
+  sed -i "/# %START%/a \"${target_to_add}\"," ${BUILD_FILE_ABS_PATH}
```
Comment (vmax) [inline tools/build_file_generator/third_party_deps/third_party_deps_updater.sh]: @val-fed please see the comment on `sed` portability above

File: tools/build_file_generator/third_party_deps/third_party_deps_updater.sh (hunk 3 above)
Comment (val-fed) [reply tools/build_file_generator/third_party_deps/third_party_deps_updater.sh]: This variable is never used, I forgot to remove it. 
Removed this line.

**Pull request merged**